woocommerce/woocommerce-android
# [WOO POS] UI tests for Product and Cart screens

Closes: #12702 


### Description

This PR adds UI tests for Product and Cart screens of POS.

<img width="605" alt="Screenshot 2024-09-27 at 8 36 39 AM" src="https://github.com/user-attachments/assets/b16c3561-e76e-4023-b646-524b5f07bae8">


### Testing information

Ensure the CI is happy - specifically, UI test pipeline is green.

### The tests that have been performed

1. Add UI test to verify Products screen is displayed when POS mode is entered
2. Add UI test to verify Cart item is added when product item is clicked
3. Add UI test to verify Checkout button is not displayed when the cart list is empty
4. Add UI test to verify different cases of Checkout button visibility
5. Add UI test to verify cart item removal functionality
6. Add UI test to verify clicking Checkout button transitions to Checkout screen and hides Products screen
7. Add UI test to verify clicking Checkout button transitions to Checkout screen and displays Totals screen
8. Add UI test to verify clicking Checkout button transitions to Checkout screen and displays "Collect Payment" button




## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
